### ISSUE TITLE
HTTPS connection to the private domain fails even if http.sslCAInfo is set

### ISSUE THREAD

**Issue by kazyx** [git-for-windows/git#825]
- [x] I was not able to find an [open](https://github.com/git-for-windows/git/issues?q=is%3Aopen)
  or [closed](https://github.com/git-for-windows/git/issues?q=is%3Aclosed) issue
  matching what I'm seeing
### Setup
- Which version of Git for Windows are you using? 32-bit or 64-bit? Include the
  output of `git version` as well.

64 bit

```
$ git --version
git version 2.9.2.windows.1
```
- Which version of Windows are you running? 32-bit or 64-bit?

64 bit_
- What options did you set as part of the installation? Or did you choose the
  defaults?

default
- Any other interesting things about your environment that might be related
  to the issue you're seeing?

Using Private(enterprise) root and intermediate SSL authority in the intranet.
Using a copy of `C:\Program Files\Git\mingw64\ssl\certs\ca-bundle.crt` with our root CA and intermediate CA certificates in the tail.
### Details
- Which terminal/shell are you running Git from? e.g Bash/CMD/PowerShell/other

Bash and CMD
- What commands did you run to trigger this issue? If you can provide a
  [Minimal, Complete, and Verifiable example](http://stackoverflow.com/help/mcve)
  this will help us understand the issue.

```
$ git clone https://our.domain/repository.git
```
- What did you expect to occur after running these commands?
  
  repository is successfully cloned.
- What actually happened instead?
  
  It fails when http.sslCAInfo is not set. (No problem)

```
Cloning into 'repository'...
fatal: unable to access 'https://our.domain/repository.git/': SSL certificate problem: unable to get local issuer certificate
```

It also fails when I set http.sslCAInfo which additionally contains our root CA and intermediate CA. (This should be fixed)

```
Cloning into 'repository'...
fatal: unable to access 'https://our.domain/repository.git/': SSL: no alternative certificate subject name matches target host name 'our.domain'
```

**Comment by dscho** [comment]
This MCVE leaves out critical steps, such as: what was the exact value of `http.sslCAInfo`, and: how to reproduce (i.e. verify your claim) without your root certificate.

**Comment by glatzert** [comment]
I think I know which problem @kazyx has here, because I ran into the same Problem.

We are using self signed certificates from our own CA in our Environment. This CA will not be trusted by Default (at least not from git-for-windows), despite it being trusted by the Computer.
It would be good, to have a small tool, which takes the current trusted roots and copies all of them to the ca-bundle.crt file.
Windows users (as myself) expect applications to use the internal cert stores of Windows.

We solved the Problem by replacing the certificate of the repository Server with public certs.
